### PR TITLE
Add extra logging to interop client to investigate failures.

### PR DESCRIPTION
**Pull request by anrossi**

### CODE REVIEW

File: src/tools/interop/interop.cpp [hunk 1]
```diff
@@ -960,14 +971,25 @@ QUIC_THREAD_CALLBACK(InteropTestCallback, Context)
...
+        "[ntrp] Test Stop, Server: %s, Port: %hu, Tests: 0x%x, Negotiated Alpn: %s, Passed: %s.",
+        PublicEndpoints[EndpointIndex].ServerName,
+        TestContext->Port,
+        (uint32_t)TestContext->Feature,
+        Alpn,
+        ThisTestFailed ? "false" : "true");
+
+    if (ThisTestFailed) {
+        free((void*)Alpn);
+    }
```
Comment (nibanks) [inline src/tools/interop/interop.cpp:992]: Unnecessary if check. `free` allows `NULL` pointers to be passed in.

**Review by nibanks — APPROVED** [review]

File: src/tools/interop/interop.cpp (hunk 1 above)
Comment (anrossi) [reply src/tools/interop/interop.cpp:992]: I'm not setting it to NULL anymore, since the logging function needs to read it.

Comment (nibanks) [reply src/tools/interop/interop.cpp:992]: Ok.

**Comment by nibanks** [comment]
Linux run spews all these:
```
MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
...
```
https://dev.azure.com/ms/msquic/_build/results?buildId=121874&view=logs&j=0d6e46ad-e8ae-59ce-364c-b792f0422f2b&t=36b6355a-9880-572c-2f7c-46de69c440d5

**Comment by nibanks** [comment]
> Linux run spews all these:
> 
> ```
> MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
> MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
> MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
> MsQuic->ConfigurationLoadCredential( Configuration, &CredConfig) FAILURE: 0xbebc212!!
> ...
> ```
> 
> https://dev.azure.com/ms/msquic/_build/results?buildId=121874&view=logs&j=0d6e46ad-e8ae-59ce-364c-b792f0422f2b&t=36b6355a-9880-572c-2f7c-46de69c440d5

Weird. I rename and it didn't repro. I don't see how this PR could cause this either... I'll probably just go ahead and merge.

**Pull request merged**